### DISCUSSION TITLE
Renaming a variable implemented in many files without need to open them.

### DISCUSSION THREAD

**Discussion by developemmanuel** [NvChad/NvChad#2003] [Q&A]
# Main issue
How can I rename a variable implemented in different files, because we I use the `<leader> r` + `a`

- **It works with opened buffers**: Here I try to rename a React component and it work with all the buffers where it's been implemented.

https://user-images.githubusercontent.com/112911715/236966623-3a7ce32a-4698-4ab8-9436-6870f9d944f3.mp4

- **Doesn't work when the buffers are not opened**: I can go to definitions (`gd`) even when they're close, but when I rename it doesn't change on the other files

https://user-images.githubusercontent.com/112911715/236966934-82af5ca4-7979-48d5-b8ca-c8422e3aaa70.mp4

## Second issue: Implementation command
- When I want to see the implementation (with `gi`) is not working but is defined in the core configuration. How can I fix it

https://user-images.githubusercontent.com/112911715/236967220-9a0a631e-ea81-4a9d-9500-8cafc4a00e5e.mp4

**Comment by siduck** [comment]
hi @developemmanuel, i'll look into this tomorrow!

**Reply by yohDone** [reply]
Hello!

I want to know if this is possible in 2026?

Thank you

**Reply by siduck** [reply]
@DYW972 I forgot about this! Can you make an issue for this I'll try this weekend!!
